napi_error_handling.md

```diff
@@ -0,0 +1,115 @@
+
+```C
+NODE_EXTERN const napi_extended_error_info* napi_get_last_error_info();
+```
+and the format of the ```napi_extended_error_info``` structure is as follows:
+
+```C
+struct napi_extended_error_info {
+  const char* error_message;
+  void* engine_reserved;
+  uint32_t engine_error_code;
+  napi_status error_code;
+};
+```
+
+```napi_get_last_error_info()``` returns the information for the last
+N-API call that was made and the  ```error_code``` will match the
+```napi_status``` returned by that call.  The ```error_message``` will
+be a textual representation of the error that occurred.
+
+## Exceptions
+
+Any N-API function may result in a pending JavaScript exception.  This is
+obviously the case for any function that may cause the execution of JavaScript
+but N-API specifies that an exception may be pending on return from any
+of the API functions.
+
+If the ```napi_status``` returned by a function is ```napi_ok``` then no
+exception is pending and no additional action is required.  If the
+```napi_status``` returned is anything other than ```napi_ok``` or
+```napi_pending_exception```, then you must call the following function in
+order to determine if an exception is pending or not:
+
+```C
+NODE_EXTERN napi_status napi_is_exception_pending(napi_env e, bool* result);
+```
+
+and then check the value of ```result```.  
+
+When an exception is pending you will generally follow one of two approaches.
+
+The first appoach is to do any appropriate cleanup and then return so that
+execution will return to JavaScript.  As part of the transition back to
+JavaScript the exception will be thrown at the point in the JavaScript
+code were the native method was invoked.  The behavior of most N-API calls
+is unspecified while an exception is pending, and many will simply return
+```napi_pending_exception```, so it is important to do as little as possible
+and then return to JavaScript where the exception can be handled.
+
+The second approach is to try to handle the exception. There will be cases
+were the native code can 'catch' the exception, take the appropriate action,
+and then continue.  It is recommended that you only do this in specific cases
+where you know you can safely handle the exception.  In these cases you can
+use the following function to get and clear the exception:
+
+```C
+NODE_EXTERN napi_status napi_get_and_clear_last_exception(napi_env e,
+                                                          napi_value* result);
+```
+
+On success, result will contain the reference to the exception.  If you
+determine after retrieving the exception that you cannot handle it after all,
+you can re-throw it with the following function:
+
+```C
+NODE_EXTERN napi_status napi_throw(napi_env e, napi_value error);
+```
+
+were error is the exception to be thrown.
+
+The following utility functions are also available in case your native code
+needs to throw an exception or determine if a ```napi_value``` represents an
+exception.
```

Comment:
Change "represents an exception" to "is an instance of a JavaScript `Error` object."